server.go

```diff
@@ -234,3 +237,128 @@ func (b *backend) Move(r *http.Request, dest *internal.Href, overwrite bool) (cr
 	}
 	return created, err
 }
+
+type HomeSet interface {
```

Comment:
I'm not super happy about this interface, because it's pretty easy to implement it externally outside of go-webdav, and implement it wrong by missing the fact that it needs to be XML-marshalable. To alleviate this, I've been thinking about making it impossible for external users to implement it. Usually this is done by adding an unexported func in the interface, e.g.

```go
type HomeSet interface {
    homeSet()
}
```

that way only the current package can implement it. But this wouldn't work here because the type is shared between our own packages, and the `carddav` package won't be able to implement the type.

Then I thought about a bunch of different solutions, like moving it to `internal/` and doing more tricks to ensure only us can implement it. But it's pretty ugly and not worth the trouble.

tl;dr I think it's fine and we just need to document the type here. Explain what it represents, explain that it needs to marshal to XML to describe the home set element, etc.